python/dremio/arguments/parse_arguments.py

```diff
@@ -0,0 +1,126 @@
+"""
+  Copyright (C) 2017-2021 Dremio Corporation
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+"""
+
+import argparse
+import sys
+import certifi
+
+
+class KVParser(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        setattr(namespace, self.dest, [])
```

Comment:
Do you want to try some functional programming here instead of this very imperative code? Basically, what you're trying to do here is to populate self.dest with utf-8 encoded tuples:
```
[value] -> [encoded_tuple] -> setattr
```

I don't know python but I think it can be expressed as two lines of code (unless you format it differently):
```
dest = list(map(lambda v: list(map(lambda a: a.encode('utf-8'), v.split('='))), values))
setattr(namespace, self.dest, dest)
```

There is https://pypi.org/project/ramda/ which is a python clone of https://ramdajs.com/docs/ and with it could look like:
```
pipe(
   map(pipe(
      split('='),
      lambda a,b: (v[0].encode('utf-8'), v[1].encode('utf-8')),
   )),
   list,
   lambda dest: setattr(namespace, self.dest, dest),
)(values)
```

It's just a suggestion to apply some fun(ctional) programming. No need to change your current code right now if you don't feel adventurous rn.

Reply:
I think this is a great learning opportunity and I'll give this a shot!